Runtime/codebase/WalletBase.cs

```diff
@@ -399,24 +400,31 @@ public Task AwaitWsRpcConnection()
         /// <param name="allowEmptySignatures"></param>
         /// <returns></returns>
         private static List<SignaturePubKeyPair> DeduplicateTransactionSignatures(
-            List<SignaturePubKeyPair> signatures, bool allowEmptySignatures = false)
+            List<SignaturePubKeyPair> signatures, bool allowEmptySignatures = false, bool preferNonEmptySignature = true)
```

Comment:
There is a difference, the current implementation of AllowEmptySignatures will introduce a bug in case of multisig transaction, for example if there is a transaction with 2 signers,  both are empty & we sign with second signer, the current implementation of AllowEmptySignatures  will remove the first and change the position of the second one, which will result in signature verification failed errror.

Reply:
Can you provide an example of an invalid transaction?